### ISSUE TITLE
avoid unneccessary dependencies

### ISSUE THREAD

**Issue by WernerMairl** [WernerMairl/CustomBaGet#1]
BaGet (root) project references BaGet.Azure => decouple, use something like a Plugin System OR provide smaller separate assemblies for better composing

**Comment by loic-sharma** [comment]
A plugin system seems heavy handed, I prefer the approach of separate assemblies. The main blocker seems to be the root BaGet project, as it contains quite a bit of code that isn't reusable. Here are the current assemblies:

1. BaGet.Core - The core that all other projects depend on. This should reference as few projects as possible. I'm pretty happy with the way this project is right now.
2. BaGet.Controllers - I'm still searching for a better name for this project.. maybe BaGet.Server? This project should contain the ASP.NET Core specific things that are agnostic to the database/storage/search providers. I need to move more code from the BaGet project to this project.
3. BaGet - The entry point for the default BaGet app. Ideally, this project just glues the other projects together and contains little code.
  * This project contains all of the Entity Framework providers. I'd like to experiment moving each context + migrations into their own project. 
  * There are some misc configuration extensions that should be moved to BaGet.Controllers
  * The service collection extensions also need to be split up and moved to separate projects. This may be tough to do.

I'll keep working on improving the situation here :)

**Comment by WernerMairl** [comment]
i agree with „don‘t over complicate things‘ or in other words avoiding overengineering!

i have a few other ideas in this direction, this is intended as a  brainstorming process...

**Comment by loic-sharma** [comment]
I was thinking about this problem some more. What do you think of moving everything that’s database/storage/cloud agnostic to the BaGet.Core project? This would include ASP.NET Core things, so this would slightly increase the size of BaGet.Core. However, it would greatly streamline things and we could get rid of the BaGet.Controllers project. What do you think?
